docs/guides/scaffolding.md

```diff
@@ -0,0 +1,102 @@
+
+1. Create a workspace folder:
+```bash
+mkdir my_workspace
+cd my_workspace
+```
+
+2. Setup the environment. Remember to use the Python version you have installed. Here we are using 3.10 as reference:
+```bash
+touch Pipfile && pipenv --python 3.10 && pipenv shell
+```
+
+3. Install the {{open_autonomy}} framework:
+```bash
+pip install open-autonomy
+```
+
+4. Initialize the framework to work with the remote [IPFS](https://ipfs.io) registry. This means that when the framework will be fetching a component, it will do so from the [IPFS](https://ipfs.io):
+    ```bash
+    autonomy init --remote --ipfs --author <your_author_name>
+    ```
+
+## Using the scaffold tool: step-by-step instructions
+
+First, create an empty agent using the autonomy CLI. This will create the agent directory, that will contain folders for connections, contracts, protocols and skills.
+```
+autonomy create my_agent
+cd my_agent
+```
+Now that the agent is in place, you can generate the basic structure of a [skill](https://open-aea.docs.autonolas.tech/skill/). For that, you need its [FSM](https://docs.autonolas.network/fsm/) specification. In this example, let's copy the contents of the [Hello World](https://docs.autonolas.network/hello_world_agent_service/) service FSM specification into a file called ```fsm_specification.yaml```. Put it inside your agent's directory.
+
+```
+alphabet_in:
+- DONE
+- NO_MAJORITY
+- RESET_TIMEOUT
+- ROUND_TIMEOUT
+default_start_state: RegistrationRound
+final_states: []
+label: packages.valory.skills.hello_world_abci.rounds.HelloWorldAbciApp
+start_states:
+- RegistrationRound
+states:
+- PrintMessageRound
+- RegistrationRound
+- ResetAndPauseRound
+- SelectKeeperRound
+transition_func:
+    (PrintMessageRound, DONE): ResetAndPauseRound
+    (PrintMessageRound, ROUND_TIMEOUT): RegistrationRound
+    (RegistrationRound, DONE): SelectKeeperRound
+    (ResetAndPauseRound, DONE): SelectKeeperRound
+    (ResetAndPauseRound, NO_MAJORITY): RegistrationRound
+    (ResetAndPauseRound, RESET_TIMEOUT): RegistrationRound
+    (SelectKeeperRound, DONE): PrintMessageRound
+    (SelectKeeperRound, NO_MAJORITY): RegistrationRound
+    (SelectKeeperRound, ROUND_TIMEOUT): RegistrationRound
+```
+
+Now, run the scaffold tool:
+
+```
+autonomy scaffold fsm my_skill --spec fsm_specification.yaml
+```
+
+You will see that the generated rounds, payloads and behaviours already appear with their correct names, as well as the `HelloWorldAbciApp` and its transition function. Now that you have succesfully created the skill, you can push it to the local registry. Since to fetch all packages from IPFS we had used the remote registry, we now need to reset it so we use a local one during the development:
+```
+autonomy init --local --reset --author AUTHOR
+```
+
+To use this local registry we need to create a `packages` directory at the same level as our agent. In this example, we need to create `my_workspace/packages`.
+```
+cd ..
+mkdir packages
+cd my_agent
+```
```

Comment:
This assumes the user does not have a `packages/` directory, we should have instructions for both ie if there's a existing packages dir and when there's not one